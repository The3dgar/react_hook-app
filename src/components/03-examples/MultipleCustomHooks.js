import React from "react";
import { useFetch } from "../../hooks/useFetch";
import { useCounter } from "../../hooks/useCounter";

export const MultipleCustomHooks = () => {
  const { counter, increment } = useCounter(1);
  const { data, loading } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${counter}`
  );
  const { quote, author } = !!data && data[0];


  return (
    <div>
      <h1>Breaking Bad Quotes</h1>
      {loading ? (
        <div className="alert alert-info text-center"> loading ... </div>
      ) : (
        <blockquote className="blockquote text-right">
          <p className="mb-0">{quote} </p>
          <footer className="blockquote-footer">{author}</footer>
        </blockquote>
      )}

      <button className="btn btn-primary" onClick={increment}>Next Quote</button>
    </div>
  );
};
