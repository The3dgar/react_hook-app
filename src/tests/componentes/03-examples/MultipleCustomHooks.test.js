import React from "react";
import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks";
import { useFetch } from "../../../hooks/useFetch";
import "@testing-library/jest-dom";
import { useCounter } from "../../../hooks/useCounter";

jest.mock("../../../hooks/useFetch")
jest.mock("../../../hooks/useCounter")

describe("Pruebas en MultipleCustomHook", () => {

  useCounter.mockReturnValue({
    counter: 10
  })

  test('debe de mostrar correctamente', () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null
    })

    const wrapper = shallow(<MultipleCustomHooks/>)
    expect(wrapper).toMatchSnapshot()
    
  })

  test('Debe de mostrar info', () => {
    useFetch.mockReturnValue({
      data: [{
        author: "Edgar",
        quote: "Hola mundo"
      }],
      loading: false,
      error: null
    })

    // Esto lo podriamos ver en el snap o podemos hacer busquedas en el hmtl

    const wrapper = shallow(<MultipleCustomHooks/>)
    expect(wrapper.find(".alert").exists()).toBe(false)
    expect(wrapper.find(".mb-0").text().trim()).toBe("Hola mundo")

  })
  
  
});
