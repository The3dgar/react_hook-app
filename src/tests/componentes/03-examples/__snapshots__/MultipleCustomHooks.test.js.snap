// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MultipleCustomHook debe de mostrar correctamente 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <div
    className="alert alert-info text-center"
  >
     loading ... 
  </div>
  <button
    className="btn btn-primary"
  >
    Next Quote
  </button>
</div>
`;
